### PR TITLE
Added kubctl github action

### PR DESCRIPTION
**Pull request by VenkataChalla554**
## Context
  Added kubectl   action  

## Changes proposed in this pull request
    Added kubctl  acrion can be refered by all pipelines

## Guidance to review
Test Pipeline : https://github.com/DFE-Digital/publish-teacher-training/actions/runs/9091604641

 

## Checklist

- [x] I have performed a self-review of my code, including formatting and typos
- [x] I have [cleaned the commit history](https://www.annashipman.co.uk/jfdi/good-pull-requests.html)
- [x] I have added the `Devops` label
- [x] I have attached the pull request to the trello card

### CODE REVIEW

**Comment by VenkataChalla554** [comment]
[Create setup-kubectl github action](https://trello.com/c/121NnR0n/1633-create-setup-kubectl-github-action)

File: set-kubectl/action.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+name: Install kubectl
+description: Sets up the kubectl
+
+runs:
+  using: composite
+  steps:
+    - name: Install kubectl
+      uses: azure/setup-kubectl@v4
+      with:
+        version: 'v1.26.10'
```
Comment (saliceti) [inline set-kubectl/action.yaml:10]: Can we change it to our current kubernetes version (1.27.7?)

**Comment by saliceti** [comment]
Can we add a README?

File: set-kubectl/action.yaml (hunk 1 above)
Comment (VenkataChalla554) [reply set-kubectl/action.yaml]: Fixed this

**Comment by VenkataChalla554** [comment]
> Can we add a README?

Added README

File: set-kubectl/README.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# Set Kubernetes CLI
+
+[kubectl](https://github.com/Azure/setup-kubectl)  is used to set up the kubectl command-line tool in your GitHub Actions workflow.kubectl is the Kubernetes command-line tool that allows you to run commands against Kubernetes clusters. By using this action, you can automate the process of installing and configuring kubectl in your CI/CD pipelines.
+
+
+## Example
+
+```yaml
+- name: Set kubectl
+  uses: DFE-Digital/github-actions/set-kubectl@1633-create-setup-kubectl-github-action
```
Comment (saliceti) [inline set-kubectl/README.md:10]: `@1633-create-setup-kubectl-github-action` -> `@master`

File: set-kubectl/README.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# Set Kubernetes CLI
+
+[kubectl](https://github.com/Azure/setup-kubectl)  is used to set up the kubectl command-line tool in your GitHub Actions workflow.kubectl is the Kubernetes command-line tool that allows you to run commands against Kubernetes clusters. By using this action, you can automate the process of installing and configuring kubectl in your CI/CD pipelines.
```
Comment (saliceti) [inline set-kubectl/README.md:3]: Add that it guarantees the kubectl version is the same as the Kubernetes cluster version

Comment (saliceti) [inline set-kubectl/README.md:3]: Maybe link to https://kubernetes.io/docs/reference/kubectl/ instead

File: set-kubectl/README.md (hunk 2 above)
Comment (VenkataChalla554) [reply set-kubectl/README.md]: Fixed

File: set-kubectl/README.md (hunk 3 above)
Comment (VenkataChalla554) [reply set-kubectl/README.md]: updated link .

File: set-kubectl/README.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
+# Set Kubernetes CLI
+
+[kubectl](https://kubernetes.io/docs/reference/kubectl/)  is used to set up the kubectl command-line tool in your GitHub Actions workflow.kubectl is the Kubernetes command-line tool that allows you to run commands against Kubernetes clusters. By using this action, you can automate the process of installing and configuring kubectl in your CI/CD pipelines.
```
Comment (saliceti) [inline set-kubectl/README.md:3]: Missing space after ".":
workflow.kubectl -> workflow. kubectl

Comment (VenkataChalla554) [reply set-kubectl/README.md]: Added missing  space

**Review by saliceti — APPROVED** [review]

File: set-kubectl/README.md [hunk 5]
```diff
@@ -0,0 +1,12 @@
+# Set Kubernetes CLI
+
+[kubectl](https://kubernetes.io/docs/reference/kubectl/)  is used to set up the kubectl command-line tool in your GitHub Actions workflow. kubectl is the Kubernetes command-line tool that allows you to run commands against Kubernetes clusters. By using this action, you can automate the process of installing and configuring kubectl in your CI/CD pipelines.
```
Comment (temitope777) [inline set-kubectl/README.md:3]: I think interact with cluster may be a better expression

**Pull request merged**